### PR TITLE
Fix ideal loads heating sizing with outdoor air

### LINKED ISSUE #7375: Ideal loads heating capacity sizing does not account for OA load - was ok in v8.9
Issue overview
--------------
User file has two identical zones with ideal loads air systems. One zone has OA, and the other does not. The cooling capacity sizing reflects the OA load, but the heating capacity sizing is identical in the two zones.  User notes that this worked OK in v8.9.

Reproduced the problem using modified PurchAirTables.idf with and without OA in zone 1.

I cannot see a workaround for this.

### Details
Some additional details for this issue (if relevant):
 - Version of EnergyPlus 9.1.0 and 9.2.0-https://github.com/NREL/EnergyPlus/commit/08197075fefb6ada0deae2e3489e55a544b98c82
 - helpdesk ticket number #15219

### Checklist
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [x] Defect file added to EnergyPlusDevSupport\DefectFiles
 - [ ] Ticket added to Pivotal for defect (development team task)
 - [ ] Pull request created (the pull request will have additional tasks related to reviewing changes that fix this defect)

### PR DESCRIPTION
**Pull request by mjwitte**
Pull request overview
---------------------
Fixes #7375. ZoneHVAC:IdealLoadsAirSystem sizing for Design Size Maximum Sensible Heating Capacity did not account for the design outdoor air flow in v9.0 and v9.1. 

### Work Checklist
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [x] Title of PR should be user-synopsis style (clearly understandable in a standalone changelog context)
 - [x] At least one of the following appropriate labels must be added to this PR to be consumed into the changelog:
   - Defect: This pull request repairs a github defect issue.  The github issue should be referenced in the PR description
   - Refactoring: This pull request includes code changes that don't change the functionality of the program, just perform refactoring
   - NewFeature: This pull request includes code to add a new feature to EnergyPlus
   - Performance: This pull request includes code changes that are directed at improving the runtime performance of EnergyPlus
   - DoNoPublish: This pull request includes changes that shouldn't be included in the changelog

### Review Checklist
This will not be exhaustively relevant to every PR.
 - [x] Functional code review (it has to work!)
 - [x] If defect, results of running current develop vs this branch should exhibit the fix
 - [x] CI status: all green or justified
 - [x] Performance: CI Linux results include performance check
 - [x] Unit Test(s)
 - C++ checks:
   - [ ] Argument types
   - [ ] If any virtual classes, ensure virtual destructor included, other things
 - IDD changes n/a
 - [ ] If new idf included, locally check the err file and other outputs

### CODE REVIEW

**Comment by mjwitte** [comment]
Develop - Before this fix - See how the cooling capacity increases with outdoor air, but not the heating capacity:
```
PurchAirTables-NoOA
 Component Sizing Information, ZoneHVAC:IdealLoadsAirSystem, ZONE1AIR, Design Size Maximum Sensible Heating Capacity [W], 7623.67810
 Component Sizing Information, ZoneHVAC:IdealLoadsAirSystem, ZONE1AIR, Design Size Maximum Total Cooling Capacity [W], 7799.32092

PurchAirTables-WithOA
 Component Sizing Information, ZoneHVAC:IdealLoadsAirSystem, ZONE1AIR, Design Size Maximum Sensible Heating Capacity [W], 7623.67810
 Component Sizing Information, ZoneHVAC:IdealLoadsAirSystem, ZONE1AIR, Design Size Maximum Total Cooling Capacity [W], 10474.86656

Ticket15219
 Component Sizing Information, ZoneHVAC:IdealLoadsAirSystem, MECHVENTXCHECKED:ZONE1 IDEAL LOADS AIR, Design Size Maximum Sensible Heating Capacity [W], 8171.70393
 Component Sizing Information, ZoneHVAC:IdealLoadsAirSystem, MECHVENTXCHECKED:ZONE1 IDEAL LOADS AIR, Design Size Maximum Total Cooling Capacity [W], 19854.26280
 Component Sizing Information, ZoneHVAC:IdealLoadsAirSystem, MECHVENTXNOTCHECKED:ZONE1 IDEAL LOADS AIR, Design Size Maximum Sensible Heating Capacity [W], 8171.70393
 Component Sizing Information, ZoneHVAC:IdealLoadsAirSystem, MECHVENTXNOTCHECKED:ZONE1 IDEAL LOADS AIR, Design Size Maximum Total Cooling Capacity [W], 13909.07795
```


After - this branch - now both heating and cooling capacity increase with outdoor air:
```
PurchAirTables-NoOA
 Component Sizing Information, ZoneHVAC:IdealLoadsAirSystem, ZONE1AIR, Design Size Maximum Sensible Heating Capacity [W], 7623.67810
 Component Sizing Information, ZoneHVAC:IdealLoadsAirSystem, ZONE1AIR, Design Size Maximum Total Cooling Capacity [W], 7799.32092

PurchAirTables-WithOA
 Component Sizing Information, ZoneHVAC:IdealLoadsAirSystem, ZONE1AIR, Design Size Maximum Sensible Heating Capacity [W], 11703.89179
 Component Sizing Information, ZoneHVAC:IdealLoadsAirSystem, ZONE1AIR, Design Size Maximum Total Cooling Capacity [W], 10474.86656

Ticket15219
 Component Sizing Information, ZoneHVAC:IdealLoadsAirSystem, MECHVENTXCHECKED:ZONE1 IDEAL LOADS AIR, Design Size Maximum Sensible Heating Capacity [W], 20559.71746
 Component Sizing Information, ZoneHVAC:IdealLoadsAirSystem, MECHVENTXCHECKED:ZONE1 IDEAL LOADS AIR, Design Size Maximum Total Cooling Capacity [W], 19854.26280
 Component Sizing Information, ZoneHVAC:IdealLoadsAirSystem, MECHVENTXNOTCHECKED:ZONE1 IDEAL LOADS AIR, Design Size Maximum Sensible Heating Capacity [W], 8171.70393
 Component Sizing Information, ZoneHVAC:IdealLoadsAirSystem, MECHVENTXNOTCHECKED:ZONE1 IDEAL LOADS AIR, Design Size Maximum Total Cooling Capacity [W], 13909.07795

File: src/EnergyPlus/PurchasedAirManager.cc [hunk 1]
```diff
@@ -1737,6 +1737,7 @@ namespace PurchasedAirManager {
                 SizingString = PurchAirNumericFields(PurchAirNum).FieldNames(FieldNum) + " [m3/s]";
                 IsAutoSize = false;
                 PrintFlag = true;
+                ZoneEqSizing(CurZoneEqNum).OAVolFlow = FinalZoneSizing(CurZoneEqNum).MinOA;
```
Comment (mjwitte) [inline src/EnergyPlus/PurchasedAirManager.cc]: Minimally invasive surgery here. Hopefully no unexpected diffs will show up.

File: tst/EnergyPlus/unit/PurchasedAirManager.unit.cc [hunk 2]
```diff
@@ -149,23 +149,20 @@ TEST_F(EnergyPlusFixture, SizePurchasedAirTest_Test1)
     DataEnvironment::StdRhoAir = 1.0; // Prevent divide by zero in ReportSizingManager
     ZoneEqSizing(CurZoneEqNum).SizingMethod.allocate(24);
     CurSysNum = 0;
-    ZoneHVACSizing.allocate(1);
-    ZoneHVACSizing(1).CoolingSAFMethod = SupplyAirFlowRate;
-    ZoneHVACSizing(1).HeatingSAFMethod = SupplyAirFlowRate;
```
Comment (mjwitte) [inline tst/EnergyPlus/unit/PurchasedAirManager.unit.cc:154]: Cleaned out some unnecessary things in the existing unit test.

File: tst/EnergyPlus/unit/PurchasedAirManager.unit.cc [hunk 3]
```diff
@@ -149,23 +149,20 @@ TEST_F(EnergyPlusFixture, SizePurchasedAirTest_Test1)
...
 
-    ZoneEqSizing(CurZoneEqNum).AirVolFlow = 0.0;
     FinalZoneSizing.allocate(1);
+    FinalZoneSizing(CurZoneEqNum).MinOA = 0.0;
+    FinalZoneSizing(CurZoneEqNum).OutTempAtHeatPeak = 5.0;
     FinalZoneSizing(CurZoneEqNum).DesHeatVolFlow = 1.0;
-    ZoneEqSizing(CurZoneEqNum).HeatingAirVolFlow = 1.0;
     FinalZoneSizing(CurZoneEqNum).DesHeatCoilInTemp = 30.0;
```
Comment (mjwitte) [inline tst/EnergyPlus/unit/PurchasedAirManager.unit.cc:157]: The unit test shouldn't be setting ZoneEqSizing - the component sizing function should be doing that.

File: tst/EnergyPlus/unit/PurchasedAirManager.unit.cc [hunk 4]
```diff
@@ -191,22 +187,65 @@ TEST_F(EnergyPlusFixture, SizePurchasedAirTest_Test1)
...
-    UnitarySysEqSizing.allocate(10);
-
     SizePurchasedAir(PurchAirNum);
     EXPECT_DOUBLE_EQ(1.0, PurchAir(PurchAirNum).MaxHeatVolFlowRate);
     EXPECT_NEAR(50985.58, PurchAir(PurchAirNum).MaxHeatSensCap, 0.1);
     EXPECT_DOUBLE_EQ(2.0, PurchAir(PurchAirNum).MaxCoolVolFlowRate);
     EXPECT_NEAR(30844.14, PurchAir(PurchAirNum).MaxCoolTotCap, 0.1);
+}
+
+TEST_F(EnergyPlusFixture, SizePurchasedAirTest_Test2)
```
Comment (mjwitte) [inline tst/EnergyPlus/unit/PurchasedAirManager.unit.cc:197]: Added a second unit test with OA.

**Review by rraustad — APPROVED** [review]
Correct use of ZoneEqSizing to pass OA to sizing routine (RequestSizing). Unit test added to include impact of OA on sizing.

**Comment by rraustad** [comment]
CI is clean, unit test confirms correction to heating coil sizing when OA > 0.

**Pull request merged**